File: rayt-rust/src/ray.rs
use crate::vec3::{Point3, Vec3};

pub(crate) struct Ray {
    pub(crate) origin: Point3,
    pub(crate) direction: Vec3,
}

impl Ray {
    pub(crate) fn new(origin: Point3, direction: Vec3) -> Self {
        Self { origin, direction }
    }

    pub(crate) fn at(&self, t: f64) -> Point3 {
        self.origin + t * self.direction
    }
}

File: rayt-python.old/rayt_python/ray.py
from rayt_python.vec3_types import Point3, Vec3


class Ray:
    def __init__(self, origin: Point3, direction: Vec3) -> None:
        self.origin = origin
        self.direction = direction

    def at(self, t: float) -> Point3:
        return self.origin + self.direction * t
